guardian/pkg/asyncutil/async.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asyncutil
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CommandExecutor[C any, R any] interface {
+	Send(C) <-chan Result[R]
```

Comment:
I saw that in most cases, we call this as Send(nill). Do we really need parameter here ?

Reply:
I'd thought about this some more, and I'd like to just stick with having the parameter here. I could make another interface that doesn't take in a parameter, but it seems like that would be unnecessary given that you just pass nil as the parameter as opposed to having different interfaces to work with.

This is intended to be a general use interface as well, and plan on moving it to somewhere more general. This will at least be used for voltron.